packages/agreement-process/src/services/readModelService.ts

```diff
@@ -289,106 +296,135 @@ async function searchTenantsByName(
 }
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function readModelServiceBuilder(
-  readModelRepository: ReadModelRepository
-) {
-  const agreements = readModelRepository.agreements;
-  const eservices = readModelRepository.eservices;
-  const tenants = readModelRepository.tenants;
-  const attributes = readModelRepository.attributes;
-  return {
-    async getAgreements(
-      filters: AgreementQueryFilters,
-      limit: number,
-      offset: number
-    ): Promise<ListResult<Agreement>> {
-      const aggregationPipeline = [
-        getAgreementsFilters(filters),
+function getDelegateAgreementsFilters(producerIds: TenantId[] | undefined) {
+  return producerIds && producerIds.length > 0
+    ? [
         {
           $lookup: {
-            from: "eservices",
+            from: "delegations",
             localField: "data.eserviceId",
-            foreignField: "data.id",
-            as: "eservices",
+            foreignField: "data.eserviceId",
+            as: "delegations",
           },
         },
         {
-          $unwind: "$eservices",
+          $unwind: {
+            path: "$delegations",
+            preserveNullAndEmptyArrays: true,
+          },
         },
-        ...(filters.showOnlyUpgradeable
-          ? [
+        {
+          $match: {
+            $or: [
               {
-                $addFields: {
-                  currentDescriptor: {
-                    $filter: {
-                      input: "$eservices.data.descriptors",
-                      as: "descr",
-                      cond: {
-                        $eq: ["$$descr.id", "$data.descriptorId"],
-                      },
-                    },
+                $and: [
+                  {
```

Comment:
Also add a filter on the delegation kind

```suggestion
                $and: [
                  {
                     "delegations.data.kind": delegationKind.delegatedProducer,
                  },
                  {
```